lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelFailedTest.java

```diff
@@ -1,8 +1,60 @@
 package io.ably.lib.test.realtime;
 
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.ably.lib.realtime.AblyRealtime;
+import io.ably.lib.realtime.Channel;
+import io.ably.lib.realtime.ChannelState;
+import io.ably.lib.realtime.ChannelStateListener;
+import io.ably.lib.realtime.ConnectionState;
+import io.ably.lib.realtime.ConnectionStateListener;
+import io.ably.lib.test.common.Helpers;
+import io.ably.lib.test.common.Setup;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.ClientOptions;
+import io.ably.lib.types.ErrorInfo;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
 /**
  * Created by kostiantyn on 8/9/2016.
  */
 
 public class RealtimeChannelFailedTest {
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        Setup.getTestVars();
+    }
+
+    @AfterClass
+    public static void tearDownAfterClass() throws Exception {
+        Setup.clearTestVars();
+    }
+
+    @Test
+    public void channel_filed_state_test(){
```

Comment:
Yes it is typo, about Channel state, there is one possible way to set channel in different state is to call "onChannelMessage" method in Channel class, all other method in Channel class like "setState", "setFailed" etc can not be called outside the class instance, by the way the "ChannelState state" member of the Channel class is a public I think it is wrong implementation, and "ChannelState state" should be private.